central/administration/events/datastore/singleton.go

```diff
@@ -0,0 +1,30 @@
+package datastore
+
+import (
+	"github.com/stackrox/rox/central/administration/events/datastore/internal/search"
+	pgStore "github.com/stackrox/rox/central/administration/events/datastore/internal/store/postgres"
+	"github.com/stackrox/rox/central/administration/events/datastore/internal/writer"
+	"github.com/stackrox/rox/central/globaldb"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	once sync.Once
+
+	ds DataStore
+)
+
+// Singleton returns a datastore instance to handle notifications.
+func Singleton() DataStore {
+	if !features.AdministrationEvents.Enabled() {
```

Comment:
Not sure I'm a fan of this being here. It seems like it could lead to mistakes. It'd be better to wrap all uses of the singleton instead